editor/editor_node.cpp

```diff
@@ -6845,7 +6848,9 @@ EditorNode::EditorNode() {
 		DisplayServer::get_singleton()->cursor_set_custom_image(Ref<Resource>());
 	}
 
-	singleton = this;
+	if (!singleton) {
+		singleton = this;
+	}
```

Comment:
Do we support constructing two EditorNode instances?

Reply:
There is no reason to, no.

Reply:
It was considered to make such checks a standard (see #79981)
Though it's rarely necessary/useful IMO.

Reply:
I added the checks to match how I've seen singleton pointers managed in other cases, future-proofing for cases like the docs generation instantiating the class. However, given that currently that's not the case, I'll just add a check.

Reply:
Done. By the way, I think this is a legit use of `DEV_ASSERT`.

Reply:
To know if this is a legit use of `DEV_ASSERT`, we need to be able to say whether only engine devs can trigger that bug. So if EditorNode can be constructed from GDScript, C# or GDExtension, this could still lead to issues. If not, `DEV_ASSERT` seems good.

Reply:
`EditorNode` is not exposed to the API.